remotes/handlers.go

```diff
@@ -62,19 +62,25 @@ func MakeRefKey(ctx context.Context, desc ocispec.Descriptor) string {
 			return prefix + "-" + desc.Digest.String()
 		}
 	}
+	key := desc.Digest.String()
+	if desc.Annotations != nil {
+		if name, ok := desc.Annotations[ocispec.AnnotationRefName]; ok {
+			key = fmt.Sprintf("%s-%s", desc.Digest.String(), name)
```

Comment:
Should this key also be used before the above return?

Also a nit on the format, `tag@digest` is preferable to `digest-tag`, the `-` isn't really anything special in terms of formatting

Reply:
Fixed